### PR TITLE
fix(reactions): allow reactions on file and image messages AP-300

### PR DESCRIPTION
**Pull request by Jekrimo**
**What this PR does** 📖
Fixes attaching reaction messages to files and image messages

**Which issue(s) this PR fixes** 🔨
AP-300

### CODE REVIEW

File: utilities/Messaging.ts [hunk 1]
```diff
@@ -96,14 +98,14 @@ export function groupMessages(
     // Checks if the message must be included in a new group
     const isNewGroup =
       groupedMessages.length === 0 ||
-      groupOrDivider?.type === 'divider' ||
+      groupOrDivider?.type === MessagingTypesEnum.DIVIDER ||
       !isSameSender ||
-      (prevMessage && !isSameGroup)
-
+      (prevMessage && !isSameGroup) ||
+      currentMessage.type === MessagingTypesEnum.FILE
```
Comment (josephmcg) [inline utilities/Messaging.ts:104]: I think I fixed this issue in my PR without forcing a new group. You should probably remove the File type check on 104.
<https://github.com/Satellite-im/Core-PWA/pull/918>

File: utilities/Messaging.ts [hunk 2]
```diff
@@ -157,36 +159,37 @@ export function updateMessageTracker(
...
+      case MessagingTypesEnum.REACTION:
         const reaction: ReactionMessage = currentMessage
         reactionsTracker[reaction.reactedTo] =
           reactionsTracker[reaction.reactedTo] || []
         if (
           !reactionsTracker[reaction.reactedTo].some(
-            (elm) => elm.id === reaction.id,
+            (elm) => {
+              return elm.id === reaction.id
+            },
```
Comment (josephmcg) [inline utilities/Messaging.ts:177]: more concise before

**Review by josephmcg — CHANGES_REQUESTED** [review]
I fixed this issue in a separate PR without forcing a new group. I like what you did with the Enums though :D

File: utilities/Messaging.ts [hunk 3]
```diff
@@ -71,11 +73,11 @@ export function groupMessages(
...
       const currentAt = dayjs(currentMessage.at)
       const prevAt = dayjs(prevMessage.at)
-      if (!dayjs().isSame(currentAt, 'day')) {
-        isSameGroup = currentAt.isSame(prevAt, 'day')
+      if (!dayjs().isSame(currentAt, MeasurementUnitsEnum.DAY)) {
+        isSameGroup = currentAt.isSame(prevAt, MeasurementUnitsEnum.DAY)
       } else {
         const prevAt = dayjs(prevMessage.at)
-        isSameGroup = currentAt.diff(prevAt, 'minutes') < 15 ? true : false
+        isSameGroup = currentAt.diff(prevAt, MeasurementUnitsEnum.MINUTES) < 15
```
[the remark this answers is not in the document]
Comment (josephmcg) [reply utilities/Messaging.ts:80]: Jeff after this fix

image

**Review by phillsatellite — APPROVED** [review]

File: utilities/Messaging.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (RetroPronghorn) [reply utilities/Messaging.ts:80]: 😂

File: store/ui/mutations.ts [hunk 4]
```diff
@@ -208,13 +208,14 @@ export default {
...
-
     if (currGroup?.messages) {
       // Find message in message group or reply in message meant for reaction
       let currMessage
       if (reaction.replyID) {
         currMessage = currGroup?.messages
-          .find((message) => message.id === reaction.messageID)
+          .find((message) => {
+            return message.id === reaction.messageID
+          })
```
[the remark this answers is not in the document]
Comment (Jekrimo) [reply store/ui/mutations.ts:218]: Ide auto format my bad

**Review by josephmcg — APPROVED** [review]
🚀

**Review by WanderingHogan — APPROVED** [review]

File: libraries/Textile/ThreadManager.ts [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+      //   threadTitle || '' + Date.now(),
+      // )
+      await this.setupDB(threadTitle  || '', threadUsers, threadID, options)
+    // }
+    // if (thread) {
+    //   if (threadTitle != null) {
+    //     const thread = await this.textile.users.getThread(threadTitle)
+    //     return ThreadID.fromString(thread.id)
+    //   }
+    // }
```
Comment (stavares843) [inline libraries/Textile/ThreadManager.ts:78]: commented out code 🔨

File: libraries/Textile/ThreadManager.ts [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+  }
+
+  async init() {
+    await this.authorize()
+
+  }
+
+  async setupDB(threadName: string, threadUsers: string[], threadID: ThreadID, options?: Object) {
+    // const thread = await this.textile.users.getThread(threadName)
+    // const threadID = ThreadID.fromString(thread.id)
```
Comment (stavares843) [inline libraries/Textile/ThreadManager.ts:37]: commented out code 🔨

File: libraries/Textile/ThreadManager.ts [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+  identity: Identity
+  threadID: ThreadID | null
+  token: string | null
+  textileClient: Client
+
+  /** @constructor
+   * Construct a ThreadManager
+   * The thread manager stores and fetches threads to store
+   * arbitrary data between two peers
+   * @argument storageMethod preffered storage method of thread assignment (LocalStorage)
```
Comment (stavares843) [inline libraries/Textile/ThreadManager.ts:18]: typo - preferred 🔨

File: libraries/Textile/ThreadManager.ts (hunk 5 above)
Comment (Jekrimo) [reply libraries/Textile/ThreadManager.ts:78]: This should not be in here, I gotta figure out how to roll this back

**Pull request merged**